### PR TITLE
Feature/dagger2

### PR DESCRIPTION
**Pull request by eunix56**
Added dagger2 to the Question Fragments in the app

### CODE REVIEW

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/repository/QuestionRepository.java [hunk 1]
```diff
@@ -6,30 +6,24 @@
...
-        this.page = page;
-        this.pageSize = pageSize;
-        this.order = order;
-        this.sortCondition = sortCondition;
-        this.site = site;
-        this.filter = filter;
-        this.siteKey = siteKey;
-        init();
+    @Inject
+    public QuestionRepository(ApiService getApiService) {
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/repository/QuestionRepository.java:21]: Change getApiService to apiService

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/MainActivity.java [hunk 2]
```diff
@@ -18,7 +19,25 @@
...
+import dagger.android.DispatchingAndroidInjector;
+import dagger.android.HasAndroidInjector;
+
+public class MainActivity extends AppCompatActivity implements HasAndroidInjector {
+
+    /*
+     * Step 1: Rather than injecting the ViewModelFactory
+     * in the activity, we are going to implement the
+     * HasActivityInjector and inject the ViewModelFactory
+     * into our MovieListFragment
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/MainActivity.java:35]: Check this comment well and make necessary correction

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByCreationFragment.java [hunk 3]
```diff
@@ -45,6 +51,23 @@
...
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        AndroidSupportInjection.inject(this);
+        super.onAttach(context);
+
+        viewModelFactory.setInputs(AppConstants.FIRST_PAGE,
+                AppConstants.PAGE_SIZE,
+                AppConstants.ORDER_DESCENDING,
+                AppConstants.SORT_BY_ACTIVITY,
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByCreationFragment.java:65]: This should be SORT_BY_CREATION

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByHotFragment.java [hunk 4]
```diff
@@ -45,6 +51,23 @@
...
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        AndroidSupportInjection.inject(this);
+        super.onAttach(context);
+
+        viewModelFactory.setInputs(AppConstants.FIRST_PAGE,
+                AppConstants.PAGE_SIZE,
+                AppConstants.ORDER_DESCENDING,
+                AppConstants.SORT_BY_ACTIVITY,
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByHotFragment.java:65]: This should be SORT_BY_HOT

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByVoteFragment.java [hunk 5]
```diff
@@ -45,6 +51,24 @@
...
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        AndroidSupportInjection.inject(this);
+        super.onAttach(context);
+
+        viewModelFactory.setInputs(AppConstants.FIRST_PAGE,
+                AppConstants.PAGE_SIZE,
+                AppConstants.ORDER_DESCENDING,
+                AppConstants.SORT_BY_ACTIVITY,
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/QuestionsByVoteFragment.java:65]: This should be SORT_BY_VOTES

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/util/AppConstants.java [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
     String QUESTION_FILTER = "!9Z(-wwYGT";
     String ANSWER_FILTER = "!9Z(-wzu0T";
-    String API_KEY = "YOUR_API_KEY";
+    String API_KEY = "QUEuzCqtqLg4NXnEOO5qug((";
```
Comment (mayorJAY) [inline app/src/main/java/com/josycom/mayorjay/flowoverstack/util/AppConstants.java:17]: Please remove this key

Comment (eunix56) [reply app/src/main/java/com/josycom/mayorjay/flowoverstack/util/AppConstants.java:17]: I'll fix it all. Thanks for pointing it out

**Comment by eunix56** [comment]
Can you please check this out and approve it if its fine? I'll make changes to the #14 and #15 and push it. Thank you

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/repository/QuestionRepository.java (hunk 1 above)
Comment (eunix56) [reply app/src/main/java/com/josycom/mayorjay/flowoverstack/repository/QuestionRepository.java]: Okay, I have fixed it. Thanks

**Comment by eunix56** [comment]
Thank you. Will make update to the others and push them

**Pull request merged**